### ISSUE TITLE
MSBuild property to disable analysis?

### ISSUE THREAD

**Issue by tompazourek** [security-code-scan/security-code-scan#178]
I'm using the NuGet package to integrate SCS, and I'd like to disable the analysis in certain builds.

Is there some MSBuild property that I can set to switch off the analysis?

Currently, the only way I figured to do this is a conditional `PackageReference` and if I don't want to do the analysis, then I don't reference the package. But this is not ideal, because it means I need to restore it separately, and have different dependency tree in each setting. I prefer to have the same dependency tree even when certain builds don't utilize the analysis.

For other Roslyn-based analyzers, I always use the `<RunAnalyzers>false</RunAnalyzers>` to switch them off. But somehow, it doesn't seem to work for SCS for me. Maybe I'm doing something wrong?

Anyway, **is there some easy way to conditionally disable the analysis while keeping the package reference?**

Thanks

**Comment by JarLob** [comment]
SCS is a roslyn analyzer, so I don't see why it shouldn't support `RunAnalyzers` if it works with other analyzers. Did you try SCS or SCS2017? Maybe it depends on codeanalysis version...

**Comment by tompazourek** [comment]
@JarLob Thanks for responding, good to know that `RunAnalyzers` is expected to work.

I'm using the latest SCS, not the 2017 version.

I investigated a bit more since writing the issue, and I still cannot figure it out. Other analyzers are respecting the `RunAnalyzers` property correctly, it's only SCS that somehow ignores it. What's interesting is that I tried to isolate it in a sample project (I tried the WebGoat, and then I built a custom one with multiple analyzers), and on all of these, `RunAnalyzers` work correctly. That would indicate that the issue is somewhere in my solution/project that I need to pin down.

I'll still investigate further, I need to really isolate the issue. I'll respond when I figure it out, so far it looks like SCS is actually not to blame, but I don't have a definitive answer yet.
